File: practice/algorithms/search/hackerland_radio_transmiters/hackerland_radio_transmiters.py
#!/usr/bin/env python

n, k = map(int, raw_input().strip().split())
x = map(int, raw_input().strip().split())
x.sort()

cnt = 0
curr_pos = 0
while curr_pos < n:
    prev_pos = curr_pos
    while curr_pos < n and x[curr_pos] - x[prev_pos] <= k:
        curr_pos += 1
    
    curr_pos -= 1
    cnt += 1

    prev_pos = curr_pos
    while curr_pos < n and x[curr_pos] - x[prev_pos] <= k:
        curr_pos += 1
print cnt

File: practice/algorithms/search/hackerland_radio_transmiters/hackerland_radio_transmiters.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int n, k;
    cin >> n >> k;
    
    vector<int> x(n);
    for(int pos = 0; pos < n; ++pos)
    {
       cin >> x[pos];
    }
    
    sort(x.begin(), x.end());

    int cnt = 0;
    int curr_pos = 0, prev_pos = 0;
    while (curr_pos < n)
    {
        prev_pos = curr_pos;
        while (curr_pos < n && (x[curr_pos] - x[prev_pos]) <= k)
            curr_pos++;
        
        curr_pos--;
        cnt++;
        
        prev_pos = curr_pos;
        while (curr_pos < n && (x[curr_pos] - x[prev_pos]) <= k)
            curr_pos++;
    }
    
    cout << cnt << endl;

    return 0;
}

